### PR TITLE
[i18nIgnore]: add a comment for translators on the data structure in localized `nav.ts` files

### PR DESCRIPTION
**Pull request by shuuji3**
#### Description (required)

I misunderstood how to update `nav.ts` for Japanese (ref. https://github.com/withastro/docs/pull/6985#issuecomment-1961828452). I also saw the same error in French translation.

To prevent incorrect data structure in `nav.ts` in the future, This adds a comment for translations in `en/nav.ts`.

/cc @Yan-Thomas 

#### Related issues & labels (optional)

- Closes #
- Suggested label:

### CODE REVIEW

File: src/i18n/en/nav.ts [hunk 1]
```diff
@@ -6,6 +6,17 @@
...
+ *
+ * For translators:
+ *
+ * To avoid accidental editing, please use the existing `{key: value}`
+ * structure with `NavDictionary()` in non-English `nav.ts` files. 
+ *
+ * For example, the entry in `en/nav.ts`
+ *   { text: 'Getting Started', slug: 'getting-started', key: 'getting-started' },
+ * should be converted to
+ *   'getting-started: 'Getting Started',
```
Comment (Fryuni) [inline src/i18n/en/nav.ts:18]:
```suggestion
 *   'getting-started': 'Getting Started',
```

**Review by Fryuni — APPROVED** [review]
Great addition! Thanks

File: src/i18n/en/nav.ts (hunk 1 above)
Comment (shuuji3) [reply src/i18n/en/nav.ts]: @Fryuni Thanks for reviewing and fixing this trivial typo! 😅

**Review by dreyfus92 — APPROVED** [review]
Nice addition @shuuji3, LGTM ✅

File: src/i18n/en/nav.ts [hunk 2]
```diff
@@ -6,6 +6,17 @@
...
+ * For translators:
+ *
+ * To avoid accidental editing, please use the existing `{key: value}`
+ * structure with `NavDictionary()` in non-English `nav.ts` files. 
+ *
+ * For example, the entry in `en/nav.ts`
+ *   { text: 'Getting Started', slug: 'getting-started', key: 'getting-started' },
+ * should be converted to
+ *   'getting-started': 'Getting Started',
+ * and only translate 'Getting Started' part into your language.
```
Comment (sarah11918) [inline src/i18n/en/nav.ts:19]:
```suggestion
 * For translators:
 *
 * Copy the English `key` value unchanged and translate only the `text` into your language:
*
 * `src/i18n/en/nav.ts`: `{ text: 'Getting Started', slug: 'getting-started', key: 'getting-started' },`
 * `src/i18n/ja/nav.ts`: `'getting-started': 'はじめに',`
```
Do you think something like this is clear enough?

**Review by sarah11918 — COMMENTED** [review]
Thank you for this helpful addition, @shuuji3 ! Because we always like to keep comments as short as possible, I made a suggestion that I think is still enough to be helpful? See what you think!

**Comment by sarah11918** [comment]
!coauthor

**Comment by shuuji3** [comment]
It's really clear and easy-to-understand instruction! Thanks for the improvement @sarah11918  👍🏻

**Pull request merged**